File: src/main/java/be/sch3lp/j2k/nullability/java/Validation.java
package be.sch3lp.j2k.nullability.java;

public final class Validation {

    private Validation() {
        // noop utility class
    }

    public static void validateNotNull(final Object value, final String field) {
        if (value == null) {
            throw new IllegalArgumentException(String.format("%s cannot be null.", field));
        }
    }

    public static void validateNullOrNotBlank(final String value, final String field) {
        if (value != null && (value.equals("") || value.equals(" "))) {
            throw new IllegalArgumentException(String.format("%s cannot be empty or blank.", field));
        }
    }

    public static void validateNotBlank(final String value, final String field) {
        validateNotNull(value, field);
        validateNullOrNotBlank(value, field);
    }
}

File: src/main/java/be/sch3lp/j2k/nullability/kotlin/Validation.kt
package be.sch3lp.j2k.nullability.kotlin

internal fun validateNotNull(value: Any?, field: String) {
    requireNotNull(value) { "$field cannot be null." }
}

internal fun validateNotBlank(value: String?, field: String) {
    require(value?.isNotBlank() ?: true) { "$field cannot be empty or blank." }
}